### PR TITLE
WIP: Solutions for Problem Set 4

### PR DESCRIPTION
**Pull request by rmohnani**

### CODE REVIEW

File: Problem_Set_4/ps4a.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    Example:
+    >>> get_permutations('abc')
+    ['abc', 'acb', 'bac', 'bca', 'cab', 'cba']
+
+    Note: depending on your implementation, you may return the permutations in
+    a different order than what is listed here.
+    '''
+    permutations = []
+
+    def insert(sequence, letter):
```
Comment (anishasthana) [inline Problem_Set_4/ps4a.py:27]: You're defining a function inside another function?

File: Problem_Set_4/ps4b.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+        for decrypting it.
+
+        Note: if multiple shifts are equally good such that they all create
+        the maximum number of valid words, you may choose any of those shifts
+        (and their corresponding decrypted messages) to return
+
+        Returns: a tuple of the best shift value used to decrypt the message
+        and the decrypted message text using that shift value
+        '''
+        # ### Initial Implementation through using a dictionary
```
Comment (anishasthana) [inline Problem_Set_4/ps4b.py]: Remove this commented code block.

**Review by anishasthana — CHANGES_REQUESTED** [review]

File: Problem_Set_4/ps4a.py (hunk 1 above)
Comment (rmohnani) [reply Problem_Set_4/ps4a.py:27]: yeah I saw them do it in one of the lectures. I had a hard time figuring out what to do here, thing started working and defining another function made everything look neater.

Comment (anishasthana) [reply Problem_Set_4/ps4a.py:27]: I mean you can still just define the function outside of this. Your call either way I guess

File: Problem_Set_4/ps4c.py [hunk 3]
```diff
@@ -152,8 +171,68 @@ def decrypt_message(self):
         
         Hint: use your function from Part 4A
         '''
-        pass #delete this line and replace with your code here
+        ### Initial Solution
```
Comment (anishasthana) [inline Problem_Set_4/ps4c.py:174]: commented stuff.

**Review by anishasthana — CHANGES_REQUESTED** [review]
remove the commented stuff

**Pull request merged**